src/CheckIfDead.php

```diff
@@ -378,15 +377,52 @@ protected function sanitizeURL( $url ) {
 			);
 		}
 		if ( isset( $parts['query'] ) ) {
-			$url .= "?".urlencode( urldecode( $parts['query'] ) );
+			$url .= "?";
+			$parts['query'] = explode( '&', $parts['query'] );
```

Comment:
Since the by the if statement, the URL has a query string, We need to add the start of the query, namely the question mark.  $parts['query'] doesn't have that.

This line breaks apart of the values it's passing to better encode the proper parts of the query.